xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -1113,6 +1138,270 @@ public String toString() {
     }
   }
 
+  static StructOrError<HttpFault> decodeFaultFilterConfig(Any rawFaultFilterConfig) {
+    if (rawFaultFilterConfig.getTypeUrl().equals(
+        "type.googleapis.com/envoy.config.filter.http.fault.v2.HTTPFault")) {
+      rawFaultFilterConfig = rawFaultFilterConfig.toBuilder().setTypeUrl(
+          "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault").build();
+    }
+    if (rawFaultFilterConfig.getTypeUrl().equals(
```

Comment:
This if condition is useless. If the type url is not as expected, it will just throw.